aten/src/ATen/native/TensorShape.cpp

```diff
@@ -587,6 +587,14 @@ Tensor narrow(const Tensor& self, int64_t dim, int64_t start, int64_t length) {
   return at::slice(self, dim, start, start + length, 1);
 }
 
+Tensor narrow(const Tensor& self, int64_t dim, const Tensor& start, int64_t length) {
+  TORCH_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
+  TORCH_CHECK(start.numel() == 1 && isIntegralType(start.scalar_type(), /*includeBool=*/false),
```

Comment:
make the error message the same as the thing you check -- i.e. "1-element integral tensor".

You can probably not mention "int" because that will be checked in the other code path.

Reply:
does the current python index handling allow any integer type?

Reply:
Updated: current python index handling allow any integer type 0-dim tensor.